src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -131,76 +129,75 @@ public ExpenseOD editExpenseInMonth(ExpenseID expenseIDEdit) throws EntityExcept
+                //Expense value less than 0 fail: When a user tries to edit the expense value with a negative number.
+                ExpenseOD expenseODFailEdit = new ExpenseOD("Expense value can't be less than $0. Please try again!");
                 return expenseOB.fail(expenseODFailEdit);
-            }//&&!expenseID.getIsRecurringExpense()
+            }
             if(!Objects.equals(expenseIDEdit.getName(), expenseIDEdit.getOldExpense())){
-                if(checkHaveSameNameInList(monthExpenseList)) {
-                    //1. Repeated name fail
+                if(checkHaveSameNameInList(monthExpenseList, expenseID.getName())) {
+                    // Repeated name fail: When a user tries to edit the expense name to a existing expense in month.
                     ExpenseOD expenseODFailEdit = new ExpenseOD("There is already a expense with this new name in this month.");
                     return expenseOB.fail(expenseODFailEdit);
                 }}
-            if (changeInRecurringInfo()) {// 4. Same recurring expense name fail
                 if (expenseID.getIsRecurringExpense()) {
-                    if (checkHaveSameNameInList(recurringExpenseList)) {
+                    //Repeated recurring expense name fail: When a user tries to edit the current expense to a recurring expense but the name already exist as a recurring expense.
+                    if (checkHaveSameNameInList(recurringExpenseList,expenseID.getName())) {
                         ExpenseOD expenseODFailEdit = new ExpenseOD("There is a recurring expense with this name, you don't need to add recurring expense in month! " +
                                 "(If this is not the same expense, please use another name!)");
                         return expenseOB.fail(expenseODFailEdit);
                     }else {
                         selectedExpense.setName(expenseIDEdit.getName());
                         expenseIDEdit.setValue(valueDouble);
                         selectedExpense.setValue(valueDouble);
-                        session.deleteRecurExpense(expenseID.getName());
+                        session.addRecurExpense(selectedExpense);
                         selectedExpense.setCategory(selectedCategory);
-                        ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this expense to the category selected in current month!");
-                        return expenseOB.success(expenseODSuccessEditRecurring);
-                    }
-                }else {
+                        // Success edit to new recurring expense.
+                        ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this new recurring expense!");
+                        return expenseOB.success(expenseODSuccessEditRecurring);}
+                }else if(checkHaveSameNameInList(recurringExpenseList, expenseID.getOldExpense())&& !expenseIDEdit.getIsRecurringExpense()) {
```

Comment:
You can just have an else statement here since we know that isRecurring is false and it used to be true (because we are keeping the if change in recurring statement), all we have to do is modify the changes and delete the expense from recurring.

Reply:
Oh yes